proposals/0000-package-dependency-import-syntax.md

```diff
@@ -0,0 +1,159 @@
+# Inline package dependency import syntax
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Rahul Malik](https://github.com/rahul-malik), [Ankit Aggarwal](https://github.com/aciidb0mb3r), [David Hart](https://github.com/hartbit)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Developers today are using Swift as a scripting language to utilize its language features and portability across macOS and Linux. Compared to other common scripting languages (Python, Ruby, etc.), Swift maintains a "batteries-not-included" approach and relies on external packages to provide functionality like command-line argument parsing that are common in scripts. This proposal focuses on a new syntax for importing package dependencies directly within scripts written in Swift with the same level of specificity offered by the current Package manifest schema.
+
+Swift-evolution thread: [Discussion thread topic for that
+proposal](https://forums.swift.org/)
+
+## Motivation
+
+Writing scripts in Swift currently requires more work than other languages. The community has filled many of the gaps in terms of functionality like command-line argument parsing but even utilizing these packages can add significantly more friction.
+
+The community has also tried to solve this problem themselves through projects like [Marathon](http://github.com/johnsundell/marathon) and [swift-sh](http://github.com/mxcl/swift-sh) to provide functionality like resolving and fetching dependencies which are available within SwiftPM today.
+There are drawbacks to this solution
+- Additional tooling is required for any host machine that runs a Swift script
+- Package resolution and fetching is performed outside of SwiftPM and can't leverage it's capabilities
+
+This proposal would dramatically improve the developer experience for writing scripts in Swift. Dependencies for common use-cases would be easy to integrate and scripts would be portable across machines without additional tools installed on the host machine. 
+
+## Proposed solution
+
+
+Improving dependency management and portabilty of scripts requires declaring dependencies within the script itself. We would implement a new import syntax for external packages. These external packages can be referenced using the `@package(url: ....)` annotation to specify the package location and version requirements.
+
+```swift
+// MyScript.swift
+import Foundation
+import UIKit
+@package(url: "http://github.com/author/example.git") import Example
+```
+
+This script can be executable by using `swift run MyScript.swift` which will utilize Swift package managers dependency fetching and resolution logic.
+
+The `@package(...)` syntax will contain the same level of control and specificity as `PackageDependencyDescription.Requirement` which should feel like a natural to developers familiar with Swift Package Manager manifest syntax.  
```

Comment:
Maybe add reference to this page: https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#package-dependency